pkg/ast/ast.go

```diff
@@ -133,6 +151,32 @@ type ContextActionRule struct {
 	TypePattern *Identifier
 	Where       *WhereClause
 }
+
+func (a *ContextActionRule) String() string {
+	var out bytes.Buffer
+	out.WriteString("{")
+	if !types.IsNilInterface(a.Context) {
+		out.WriteString(fmt.Sprintf("Context: %s, ", a.Context.String()))
```

Comment:
these should be in the same seal syntax as the input

Reply:
done